integrated_channels/degreed/client.py

```diff
@@ -0,0 +1,213 @@
+# -*- coding: utf-8 -*-
+"""
+Client for connecting to Degreed.
+"""
+
+from __future__ import absolute_import, unicode_literals
+
+import datetime
+import time
+
+import requests
+
+from django.apps import apps
+
+from six.moves.urllib.parse import urljoin  # pylint: disable=import-error
+
+
+class DegreedAPIClient(object):
+    """
+    Client for connecting to Degreed.
+
+    Specifically, this class supports obtaining access tokens and posting to the courses and
+    completion status endpoints.
+    """
+
+    CONTENT_PROVIDER_SCOPE = 'provider_content'
+    COMPLETION_PROVIDER_SCOPE = 'provider_completion'
+    SESSION_TIMEOUT = 60
+
+    def __init__(self, enterprise_configuration):
+        """
+        Instantiate a new client.
+
+        Args:
+            enterprise_configuration (DegreedEnterpriseCustomerConfiguration): An enterprise customers's
+            configuration model for connecting with Degreed
+
+        Raises:
+            ValueError: If a URL or access token are not provided.
+        """
+
+        if not enterprise_configuration:
+            raise ValueError('An DegreedEnterpriseCustomerConfiguration must be supplied!')
+
+        self.global_degreed_config = apps.get_model('degreed', 'DegreedGlobalConfiguration').current()
+        self.enterprise_configuration = enterprise_configuration
+        self.session = None
+        self.expires_at = None
+
+    def send_completion_status(self, payload):
```

Comment:
Can we change the name here to `create_course_completion` to better align with the Degreed API? I know we probably won't ever need to create book completions for instance, but then again who knows.

Reply:
:+1: Good idea

Reply:
This would actually require me to change some SAPSF stuff too, btw, because a base client is calling certain methods.